### PR TITLE
Add UML for add_session command

### LINKED ISSUE #96: UML for Session
Create uml diagram for session to update in DG

### PR DESCRIPTION
**Pull request by keithxun**
fix #96

### CODE REVIEW

File: docs/diagrams/Session.puml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+activate logic LOGIC_COLOR
+
+logic -[LOGIC_COLOR]> model : getFilteredMemberList()
+activate model MODEL_COLOR
+
+model -[MODEL_COLOR]> logic : List<Member>
+deactivate model
+
+loop For each member
+    logic -[LOGIC_COLOR]> model : setMember(updatedMember, updatedMember)
```
Comment (yuvrajaryan) [inline docs/diagrams/Session.puml]: Is there a way to include the member.addSession(toAdd); component as well

File: docs/diagrams/Session.puml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    model -[MODEL_COLOR]-> logic
+    deactivate model
+end
+
+logic -[LOGIC_COLOR]> storage : saveAddressBook(addressBook)
+activate storage STORAGE_COLOR
+
+storage -[STORAGE_COLOR]> storage : Save to file
+activate storage STORAGE_COLOR_T1
+storage --[STORAGE_COLOR]> storage
```
Comment (yuvrajaryan) [inline docs/diagrams/Session.puml:35]: whats the point of the second arrow

File: docs/diagrams/Session.puml (hunk 1 above)
Comment (keithxun) [reply docs/diagrams/Session.puml]: Added

File: docs/diagrams/Session.puml (hunk 2 above)
Comment (keithxun) [reply docs/diagrams/Session.puml:35]: Indicates end of saving to storage

File: docs/diagrams/Session.puml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+!include style.puml
+skinparam ArrowFontStyle plain
+
+Actor User as user USER_COLOR
+Participant ":UI" as ui UI_COLOR
+Participant ":Logic" as logic LOGIC_COLOR
+Participant ":Model" as model MODEL_COLOR
+Participant ":Storage" as storage STORAGE_COLOR
+
+user -[USER_COLOR]> ui : "add_session Rehearsal 24 Oct 2024 2 1"
```
Comment (QinHaichen12) [inline docs/diagrams/Session.puml:11]: Should there be brackets here? Or is it the command with name tags liks s/

File: docs/diagrams/Session.puml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ui -[UI_COLOR]> logic : execute("add_session Rehearsal 24 Oct 2024 2 1")
+activate logic LOGIC_COLOR
+
+logic -[LOGIC_COLOR]> logic : createSession("Rehearsal", "24 Oct 2024", 2, 1)
+activate logic LOGIC_COLOR
+logic --[LOGIC_COLOR]> logic
+deactivate logic
+
+
+loop For each Member
```
Comment (QinHaichen12) [inline docs/diagrams/Session.puml:23]: is this the condition?

File: docs/diagrams/Session.puml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+storage --[STORAGE_COLOR]> logic
+deactivate storage
+
+logic --[LOGIC_COLOR]> ui
+deactivate logic
+
+ui--[UI_COLOR]> user
+deactivate ui
+@enduml
```
Comment (QinHaichen12) [inline docs/diagrams/Session.puml:46]: i think the author of this PR is very handsome

**Review by QinHaichen12 — COMMENTED** [review]
LGTM! Nice inclusion

**Pull request merged**